src/main/java/de/dhbw/karlsruhe/services/UserService.java

```diff
@@ -0,0 +1,65 @@
+package de.dhbw.karlsruhe.services;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import de.dhbw.karlsruhe.adapter.UserAdapter;
+import de.dhbw.karlsruhe.model.User;
+import de.dhbw.karlsruhe.ports.UserPort;
+
+public class UserService {
+
+	private final EncryptionService encryptionService = new EncryptionService();
+	private final UserPort userPort = new UserAdapter();
+
+	public void saveUser(User user) {
+		userPort.saveUser(user);
+	}
+
+	public boolean createUser(User createUser) throws NoSuchAlgorithmException {
+
+		if (isUserNameFree(createUser.getUserName()) && isPasswordNotEmpty(createUser.getPassword())
+				&& isUserNameWithoutSpecialCharacters(createUser.getUserName())
+				&& isPasswordWithoutSpecialCharacters(createUser.getPassword())) {
+			String encryptPassword = encryptionService.getSHAEncryptedPassword(createUser.getPassword());
+			saveUser(new User(createUser.getUserName(), encryptPassword));
```

Comment:
Auch überprüfen, ob Nutzername leer ist, sonst kommt es zum Absturz.

_Do you already have an account? y/n
n
Please enter a username: 
Please enter a password: al
Please enter your username: 
Please enter your password: al
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at de.dhbw.karlsruhe.adapter.UserAdapter.getPassword(UserAdapter.java:40)
	at de.dhbw.karlsruhe.services.UserService.getPasswordFromUserName(UserService.java:42)
	at de.dhbw.karlsruhe.services.UserService.isPasswordCorrect(UserService.java:36)
	at de.dhbw.karlsruhe.services.StartUpDialogService.loginUser(StartUpDialogService.java:41)
	at de.dhbw.karlsruhe.services.StartUpDialogService.login(StartUpDialogService.java:26)
	at de.dhbw.karlsruhe.Main.main(Main.java:10)_

Reply:
Angepasst